seqr/views/utils/orm_to_json_utils.py

```diff
@@ -632,70 +633,32 @@ def get_json_for_gene_notes_by_gene_id(gene_ids, user):
+        nested_fields=[{'fields': ('seqr_locus_list', 'guid'), 'key': 'locusListGuid'}]
+    )
+    pa_json_by_locus_list = {pa.pop('locusListGuid'): pa for pa in pa_json}
+    for ll in locus_lists_json:
+        pa_locus_list_json = pa_json_by_locus_list.get(ll['locusListGuid'])
+        if pa_locus_list_json:
+            ll['paLocusList'] = pa_locus_list_json
 
-    def _process_result(result, locus_list):
-        gene_set = locus_list.locuslistgene_set
-        interval_set = locus_list.locuslistinterval_set
-        if include_genes:
-            intervals = _get_json_for_models(interval_set.all())
-            genome_versions = {interval['genomeVersion'] for interval in intervals}
-
-            if include_pagenes:
-                result.update({
-                    'items': [{
-                        'geneId': gene.gene_id,
-                        'pagene': _get_json_for_model(gene.palocuslistgene, user=user, is_analyst=is_analyst)
-                        if hasattr(gene, 'palocuslistgene') else None
-                    } for gene in gene_set.all()] + intervals,
-                })
-            else:
-                result.update({
-                    'items': [{'geneId': gene.gene_id} for gene in gene_set.all()] + intervals,
-                })
-            result.update({
-                'intervalGenomeVersion': genome_versions.pop() if len(genome_versions) == 1 else None,
-            })
-
-        if include_project_count:
-            result['numProjects'] = locus_list.num_projects
-
-        if include_metadata:
-            result.update({
-                'numEntries': gene_set.count() + interval_set.count()
-                if include_genes else locus_list.locuslistgene__count + locus_list.locuslistinterval__count,
-                'canEdit': user == locus_list.created_by,
-            })
-
-        if hasattr(locus_list, 'palocuslist'):
-            pa_locus_list_json = _get_json_for_model(locus_list.palocuslist, user=user, is_analyst=is_analyst)
-            result.update({
-                'paLocusList': pa_locus_list_json,
-            })
 
+def get_json_for_locus_lists(locus_lists, user, include_metadata=True, additional_values=None):
+    ll_additional_values = {}
+    if additional_values:
+        ll_additional_values.update(additional_values)
     if include_metadata:
-        prefetch_related_objects(locus_lists, 'created_by')
-
-    if include_genes:
-        prefetch_related_objects(locus_lists, 'locuslistgene_set')
-        prefetch_related_objects(locus_lists, 'locuslistinterval_set')
-    elif include_metadata:
-        locus_lists = locus_lists.annotate(Count('locuslistgene')).annotate(Count('locuslistinterval'))
-
-    prefetch_related_objects(locus_lists, 'palocuslist')
-
-    if include_pagenes:
-        prefetch_related_objects(locus_lists, 'locuslistgene_set__palocuslistgene')
+        ll_additional_values.update({
+            'numEntries': Count('locuslistgene') + Count('locuslistinterval'),
```

Comment:
so this is not like a choice we have, its how django works. If you want to count things you have to tell django the string name of what to count. Class name churn is not at all a thing for a database ORM model - it would require a full database migration to rename a table that will involve refactoring our entire codebase